pkg/agent/manager/cache/cache.go

```diff
@@ -744,10 +955,31 @@ func sortIdentities(identities []Identity) {
 	})
 }
 
-func makeIdentity(record *cacheRecord) Identity {
+func sortEntries(entries []*common.RegistrationEntry) {
+	sort.Slice(entries, func(a, b int) bool {
+		return entries[a].EntryId < entries[b].EntryId
+	})
+}
+
+func sortTimestamps(records []record) {
+	sort.Slice(records, func(a, b int) bool {
+		return records[a].timestamp < records[b].timestamp
+	})
+}
+
+func makeIdentity(record *cacheRecord, svid *X509SVID) Identity {
 	return Identity{
 		Entry:      record.entry,
-		SVID:       record.svid.Chain,
-		PrivateKey: record.svid.PrivateKey,
+		SVID:       svid.Chain,
+		PrivateKey: svid.PrivateKey,
 	}
 }
+
+type record struct {
```

Comment:
This name is kind of confusing because it's not a cache record per se, and there's also a type called `cacheRecord` in this file/package. What about something like `recordAccessEvent`?